ext/liquid_c/vm.c

```diff
@@ -138,10 +166,6 @@ static void write_obj(VALUE output, VALUE obj)
             break;
         case T_NIL:
             break;
-        default:
-            obj = rb_funcall(obj, id_to_s, 0);
-            rb_str_append(output, obj);
```

Comment:
Who is doing this append in the new code path?

Reply:
I moved the default case to the top of this switch statement, so it falls through to `case T_STRING` after coercing the object to a string.